scripts/download-resources.mjs

```diff
@@ -80,53 +90,173 @@ export async function download(url, destPath, overwrite = false, access = fs.con
       args[0] = path.join(process.env.SystemRoot, 'system32', 'tar.exe');
     }
     spawnSync(...args);
-    fs.copyFileSync(path.join(helmDir, `${ kubePlatform }-amd64`, exeName('helm')), helmFinalPath);
-    fs.chmodSync(helmFinalPath, 0o755);
+    binaryFinalPath = path.join(binDir, exeName(binaryBasename));
+    fs.copyFileSync(path.join(tgzDir, tgzPlatformDir, exeName(binaryBasename)), binaryFinalPath);
+    fs.chmodSync(binaryFinalPath, 0o755);
+  } finally {
+    console.log('finishing...');
+    fs.rmSync(tgzDir, { recursive: true, maxRetries: 10 });
+  }
+
+  return binaryFinalPath;
+}
+
+// Download a zip file to a temp dir, expand,
+// and move the expected binary to the final dir
+async function downloadZip(url, binaryBasename, zipPlatformDir) {
+  const zipDir = fs.mkdtempSync(path.join(os.tmpdir(), `rd-${ binaryBasename }-`));
+  let binaryFinalPath = '';
+
+  try {
+    const zipPath = path.join(zipDir, `${ binaryBasename }.zip`);
+    const args = ['unzip', '-o', zipPath, '-d', zipDir];
+
+    await download(url, zipPath, false, fs.constants.W_OK);
+    spawnSync(...args);
+    binaryFinalPath = path.join(binDir, exeName(binaryBasename));
+    fs.copyFileSync(path.join(zipDir, zipPlatformDir, exeName(binaryBasename)), binaryFinalPath);
+    fs.chmodSync(binaryFinalPath, 0o755);
   } finally {
-    fs.rmSync(helmDir, { recursive: true, maxRetries: 10 });
+    console.log('finishing...');
+    fs.rmSync(zipDir, { recursive: true, maxRetries: 10 });
+  }
+
+  return binaryFinalPath;
+}
+
+/**
+ * Find the home directory, in a way that is compatible with
+ * kuberlr
+ */
+async function findHome() {
+  const tryAccess = async(path) => {
+    try {
+      await fs.promises.access(path);
+
+      return true;
+    } catch {
+      return false;
+    }
+  };
+
+  const osHomeDir = os.homedir();
+
+  if (osHomeDir && await tryAccess(osHomeDir)) {
+    return osHomeDir;
+  }
+  if (process.env.HOME && await tryAccess(process.env.HOME)) {
+    return process.env.HOME;
+  }
+  if (onWindows) {
+    if (process.env.USERPROFILE && await tryAccess(process.env.USERPROFILE)) {
+      return process.env.USERPROFILE;
+    }
+    if (process.env.HOMEDRIVE && process.env.HOMEPATH) {
+      const homePath = path.join(process.env.HOMEDRIVE, process.env.HOMEPATH);
+
+      if (await tryAccess(homePath)) {
+        return homePath;
+      }
+    }
   }
 
+  return null;
+}
+
+export default async function main() {
+  // Download Kubectl
+  const kubeVersion = (await (await fetch('https://dl.k8s.io/release/stable.txt')).text()).trim();
+  const kubectlURL = `https://dl.k8s.io/${ kubeVersion }/bin/${ kubePlatform }/amd64/${ exeName('kubectl') }`;
+  const kubectlPath = path.join(binDir, exeName('kubectl'));
+
+  fs.mkdirSync(binDir, { recursive: true });
+  // If kubectlPath is a symlink delete it before continuing
+  try {
+    const stat = await fs.promises.lstat(kubectlPath);
+
+    if (stat.isSymbolicLink()) {
```

Comment:
True. I wasn't sure if we can be certain that all the assets are immutable, as they aren't being identified with SHAs. If there's a CVE and a provider decides to update the versions in place, we would want to get those.

The main risk of re-downloading is just redundancy.

Reply:
If there's a CVE, the provider (that is, the Kubernetes release people) have historically released new patch releases?

Also, this is we _always_ re-download, not just we _may_ re-download, which is rather annoying.

Reply:
This has been rewritten -- if we're using kuberlr (and we only turn if off if it can't be downloaded) we download directly into `~/.kuberlr/PLATFORM/kubectlVERSION` -- and this is the only case where we don't redownload. But because all the other binaries end up as versionless names, or come from zip files that are stored temporarily, we download everything else possibly redundantly.